### PR TITLE
Weather Today app

### PR DESCRIPTION
**Pull request by GzimAsani**
In this project:

- I have used webpack to bundle different files
- Worked with the API of https://openweathermap.org/
- Used async and await.

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello @GzimAsani  :wave:

__Your project is looking great!__

_**Top highlights:**_
- All checks are passed successfully :100: 
- Nice PR title and summary :+1: 

__However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. `/node_modules`). This is part of the project specific requirements. Always include a rule for the `node_modules` directory on `.gitignore` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
- If you've already committed the `node_modules` directory before you added the rule to the `.gitignore` file, then you'll need to do the following:
  - Delete the `node_modules` directory
  - Commit and push your changes
  - Run `npm install` to re-install the project dependencies for local use. 
  - That's it!
- [ ] The `Install` section of your Readme is not complete: you only mentioned how the user will install the dependencies, you should talk about how they will run your application locally on their browsers.
- [ ] You've added the toggle control buttons (Celcius and Fahrenheit), kindly make them work: your app should display a unit at a time, then the buttons should be used to toggle.
- [ ] All webpack related packages should be included in package.json like `webpack`, `webpack-cli`, `webpack-dev-server` etc. Also, those packages should be added as devDependencies ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/devdep_requirement.png) from [Pathwright)](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

- [ ] Necessary run common commands like build, development should be in package.json inside `scripts'. Here is a basic example
```JSON
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  }
```

- [ ] You should also include error handling for the `async/await` process. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))

**[ OPTIONAL CHANGES ]**

- [ ] Although you've added the `appearance changing feature`, however, I personally think you should make your app cover the whole page and make the appearance change better. You can do this by changing the color of the background, or by making the images that describe the weather more bigger and visible.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by GzimAsani** [comment]
For the next reviewer, please be aware that my toggle for Celsius and Fahrenheit is working properly if you clone the repo. But for some reason it is not doing the same in the live page on netlify

File: package.json
Comment (teekaytech) [inline package.json]: **REQUIRED**
- Update the build script to: ` "build": "webpack --mode production"`
- Add the dev script: "dev": "webpack --mode development"

**OPTIONAL**
`"start": "webpack-dev-server --mode development --open"`

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello @GzimAsani :wave:

__Thank you for implementing most of the required changes.
However, you still have some outstanding changes to be made. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] As I mentioned earlier, run common commands like build and development should be in package.json inside `scripts`. Check my inline comment for details.
- [ ] Also, including error handling for the `async/await` API call is one of this project's specific requirements. You can use a simple `try...catch` block to enforce this.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by GzimAsani** [comment]
I've used catch in app.js on line 17 which is connected with my async

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work. 💻 👀 ✨ 
- Great work applying most of the changes requested by the previous TSEs.

## However, there are some changes required 👀 🧰 🔨 🔧

- Please add error handling with a `try/catch` statement in your async function in `weather.js`
- You may follow this link for more information on how to do this: https://itnext.io/error-handling-with-async-await-in-js-26c3f20bc06a

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****


 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**
